src/openms/include/OpenMS/METADATA/ID/IdentifiedCompound.h

```diff
@@ -0,0 +1,87 @@
+//  * Neither the name of any author or any participating institution
+//    may be used to endorse or promote products derived from this software
+//    without specific prior written permission.
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Hendrik Weisser $
+// $Authors: Hendrik Weisser $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_METADATA_ID_IDENTIFIEDCOMPOUND_H
+#define OPENMS_METADATA_ID_IDENTIFIEDCOMPOUND_H
+
+#include <OpenMS/METADATA/ID/ScoredProcessingResult.h>
+
+#include <boost/multi_index_container.hpp>
+#include <boost/multi_index/ordered_index.hpp>
+#include <boost/multi_index/member.hpp>
+
+namespace OpenMS
+{
+  namespace IdentificationDataInternal
+  {
+    struct IdentifiedCompound: public ScoredProcessingResult
+    {
+      String identifier;
+
+      EmpiricalFormula formula;
+
+      String name;
+
+      String smile;
+
+      String inchi;
+
+      explicit IdentifiedCompound(
+        const String& identifier,
+        const EmpiricalFormula& formula = EmpiricalFormula(),
+        const String& name = "", const String& smile = "",
+        const String& inchi = "", const ScoreList& scores = ScoreList(),
+        const std::vector<ProcessingStepRef>& processing_step_refs =
+        std::vector<ProcessingStepRef>()):
+        ScoredProcessingResult(scores, processing_step_refs),
+        identifier(identifier), formula(formula), name(name), smile(smile),
+        inchi(inchi)
+      {
+      }
+
+      IdentifiedCompound(const IdentifiedCompound& other) = default;
+    };
+
+    // identified compounds indexed by their identifiers:
+    typedef boost::multi_index_container<
+      IdentifiedCompound,
+      boost::multi_index::indexed_by<
+        boost::multi_index::ordered_unique<boost::multi_index::member<
+          IdentifiedCompound, String, &IdentifiedCompound::identifier>>>
+      > IdentifiedCompounds;
```

Comment:
I dont quite understand this, I thought the point of `boost::multi_index_container` was to index by `A` and also sort by `B` -- but here you only use the indexing part and not the second sorting part. Is that correct? Could this be replaced by a sorted list by `IdentifiedCompound::identifier` ?

Reply:
Yes, in principle it could be replaced with a set (I think), but the multi_index_container makes it easy to index by "identifier" and provides additional useful features (range retrieval, updating of elements).